src/api/observable.ts

```diff
@@ -116,6 +116,35 @@ export class IObservableFactories {
 		return res as any;
 	}
 
+	dynamic<T>(props: T, name?: string): T & IObservableObject {
+		if (arguments.length > 2)
+			incorrectlyUsedAsDecorator("object");
+		invariant(typeof Proxy === 'function', "dynamic objects are not supported in this environment");
+		const internalMap = new ObservableMap(props, referenceEnhancer, name);
+		const result = new Proxy(internalMap, {
```

Comment:
I think you could also add a `deleteProperty` and maybe a `has` trap for the sake of completeness.